chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -46,9 +46,16 @@ while test $# -gt 0; do
         -h|--help|--usage|-\?) usage ;;
         -v|--verbose) CROWBAR_VERBOSE=1 ;;
         -f|--force) CROWBAR_FORCE=1 ;;
-        -k|--keep-existing-hostname) KEEP_EXISTING_HOSTNAME=1 ;;
         --gpg-auto-import-keys) CROWBAR_AUTO_IMPORT_KEYS=1 ;;
         --no-gpg-checks) CROWBAR_NO_GPG_CHECKS=1 ;;
+        -k|--keep-existing-hostname)
+            if test $# -eq 1; then
```

Comment:
+1 for the over-engineering :). Regarding that I am still wondering why we need -k to have a value at all. I guess the idea is to address https://github.com/crowbar/barclamp-provisioner/pull/398#discussion_r25850262 with this, right?

But do we really need/want to support the case where the user can override  ```node[:provisioner][:keep_existing_hostname]``` by calling the script with  -k 0? Maybe @vuntz can clarify. Otherwise I'd say do not expose the -k option at all when ```node[:provisioner][:keep_existing_hostname]`` is ```true```